### PR TITLE
Update jupyter style guide

### LINKED ISSUE #4821: Standardize names for common variables and quantities in the Jupyter Notebook Style Guide
Standardize names for common variables and quantities in the [Jupyter Notebook Style Guide](https://github.com/pymc-devs/pymc3/wiki/PyMC3-Jupyter-Notebook-Style-Guide), so that the style is consistent throughout the documentation.

Suggestion by @OriolAbril :
> It would probably be good to always use the same names, for sample stats we defined a convention at ArviZ that now PyMC3 will use because it uses InferenceData but there are still many other similar cases that could be considered, stacking `chain, draw` dimensions ->` sample`? `__sample__`? `draws`? `chain_draw`?

### PR DESCRIPTION
**Pull request by OriolAbril**
Update the jupyter style guide to include explanation on the extra dependencies template introduced in https://github.com/pymc-devs/pymc-examples/pull/249 (already used in the [glm robust with outlier detection example](https://docs.pymc.io/projects/examples/en/latest/generalized_linear_models/GLM-robust-with-outlier-detection.html#installation-notes))

### CODE REVIEW

File: docs/source/contributing/jupyter_style.md [hunk 1]
```diff
@@ -58,6 +58,123 @@ Choose a category from [existing categories](https://github.com/pymc-devs/pymc/w
...
+        },
+        "myst_substitutions": {
+          "pip_dependencies": "graphviz",
+          "conda_dependencies": "python-graphviz",
+        }
+      }
+      ```
+
+      The pip and conda spcific keys overwrite the `extra_installs` one, so it doesn't make
+      sense to use `extra_installs` is using them. Either both pip and conda substitutions
```
Comment (michaelosthege) [inline docs/source/contributing/jupyter_style.md]: Sentence grammar is broken near "is using them"

Comment (OriolAbril) [reply docs/source/contributing/jupyter_style.md]: I think I wanted to write "if using them", but would be happy to change the sentence altogether

File: docs/source/contributing/jupyter_style.md [hunk 2]
```diff
@@ -28,6 +28,117 @@ the repository where the notebook is in (pymc or pymc-examples).
...
+* For stats and diagnostics, use the ArviZ function name as variable name: `ess = az.ess(...)`, `loo = az.loo(...)`
+* If there are multiple models in a notebook, assign a prefix to each model,
+  and use it throughout to identify which variables map to each model.
+  Taking the famous eight school as example, with a `centered` and `non_centered` model
+  to compare parametrizations, use `centered_model` (pm.Model object), `centered_idata`, `centered_post`, `centered_ess`... and `non_centered_model`, `non_centered_idata`...
+
+**Dimension and random variable names**
+* Use singular dimension names, following ArviZ `chain` and `draw`.
+  For example `sample`, `cluster`, `axis`, `component`, `forest`, `time`...
+* If you can't think of a meaningful name for the dimension representing the number of observations such as time, fall back to `obs_id`
```
Comment (ltoniazzi) [inline docs/source/contributing/jupyter_style.md]: No period at the end of lines 51, 52 and 53

File: docs/source/contributing/jupyter_style.md [hunk 3]
```diff
@@ -58,6 +169,123 @@ Choose a category from [existing categories](https://github.com/pymc-devs/pymc/w
...
+      }
+      ```
+
+      The pip and conda spcific keys overwrite the `extra_installs` one, so it doesn't make
+      sense to use `extra_installs` if using them. Either both pip and conda substitutions
+      are defined or none of them is.
+      :::
+      ::::
+
+1.  include the warning and installation advise template with the following markdown:
```
Comment (ltoniazzi) [inline docs/source/contributing/jupyter_style.md]: Is there a position preferred for this markdown, for example after the title or  after the imports?

**Review by ltoniazzi — COMMENTED** [review]
Not sure you want to include that doc type references should use paths. Mentioning this as I struggled with it in [PR 195](https://github.com/pymc-devs/pymc-examples/pull/195) (see @OriolAbril reviewed on 4 Sep there), and I could not easily reach a page explaining this from the [Jupyter Style Guide](https://github.com/pymc-devs/pymc/blob/main/docs/source/contributing/jupyter_style.md).

File: docs/source/contributing/jupyter_style.md (hunk 3 above)
Comment (OriolAbril) [reply docs/source/contributing/jupyter_style.md]: good point, I'd say right before imports so libraries can be installed and then imported

File: docs/source/contributing/jupyter_style.md [hunk 4]
```diff
@@ -28,6 +28,117 @@ the repository where the notebook is in (pymc or pymc-examples).
...
+  `post`/`posterior`, `const`/`constant_data`, `post_pred`/`posterior_predictive` or `obs_data`/`observed_data`
+* For stats and diagnostics, use the ArviZ function name as variable name: `ess = az.ess(...)`, `loo = az.loo(...)`
+* If there are multiple models in a notebook, assign a prefix to each model,
+  and use it throughout to identify which variables map to each model.
+  Taking the famous eight school as example, with a `centered` and `non_centered` model
+  to compare parametrizations, use `centered_model` (pm.Model object), `centered_idata`, `centered_post`, `centered_ess`... and `non_centered_model`, `non_centered_idata`...
+
+**Dimension and random variable names**
+* Use singular dimension names, following ArviZ `chain` and `draw`.
+  For example `sample`, `cluster`, `axis`, `component`, `forest`, `time`...
```
Comment (michaelosthege) [inline docs/source/contributing/jupyter_style.md]: IIRC "sample" is a protected variable name that PyMC will deny automatically.

**Review by michaelosthege — APPROVED** [review]
merge when you think it's a good time

**Pull request merged**